Napzak-iOS/Napzak-iOS/Global/Components/ProductItemView.swift

```diff
@@ -0,0 +1,171 @@
+                    .retry(maxCount: 3, interval: .seconds(5))
+                    .onFailure { error  in
+                        print("failure: \(error.localizedDescription)")
+                    }
+                    .resizable()
+                    .scaledToFill()
+                    .frame(width: width, height: width * 1.05)
+                    .clipped()
+            } else {
+                Rectangle()
+                    .fill(Color.napzakGrayScale(.gray100))
+                    .frame(width: width, height: width)
+            }
+            
+            if product.tradeStatus != .beforeTrade {
+                tradeStatusView
+            }
+            
+            HStack(alignment: .bottom, spacing: 2) {
+                switch product.tradeType {
+                case .sell:
+                    Image(.imgChipSell)
+                        .resizable()
+                        .frame(width: 40, height: 20)
+                case .buy:
+                    Image(.imgChipBuy)
+                        .resizable()
+                        .frame(width: 40, height: 20)
+                }
+                if let isPriceNegotiable = product.isPriceNegotiable {
+                    if isPriceNegotiable {
+                        Image(.imgChipBidding)
+                            .frame(width: 50, height: 20)
+                    }
+                }
+                Spacer()
+                if !product.isOwnedByCurrentUser {
+                    likeButton
+                }
+            }
+        }
+        .frame(width: width, height: width * 1.05)
+        .clipShape(RoundedRectangle(cornerRadius: 3))
+    }
+    
+    private var likeButton: some View {
+        Button {
+            if shouldToggleInterestState() {
+                product.isInterested.toggle()
+            }
+        } label: {
+            Image(product.isInterested ? .btnHeartSelected : .btnHeartDefault)
+                .resizable()
+                .frame(width: 14, height: 13)
+        }
+        .padding([.bottom, .trailing], 9)
+    }
+    
+    private var productInfo: some View {
+        VStack(alignment: .leading, spacing: 4) {
+            Text(product.genreName)
+                .applyNapzakFont(.caption3Regular12)
+                .foregroundStyle(Color.napzakGrayScale(.gray500))
+                .frame(height: 15)
+            Text(product.productName)
+                .applyNapzakFont(.body5SemiBold14)
+                .foregroundStyle(Color.napzakGrayScale(.gray500))
+                .frame(height: 18)
+                .lineLimit(1)
+            Text(product.tradeType == .sell ? "\(String(product.price).convertPrice(maxPrice: 1_000_000))원" : "\(String(product.price).convertPrice(maxPrice: 1_000_000))원대")
+                .applyNapzakFont(.body1Bold16)
+                .foregroundStyle(Color.napzakGrayScale(.gray500))
+                .frame(height: 20)
+                .lineLimit(1)
+                .padding(.top, 4)
```

Comment:
갑자기 떠오는 아이디어인데요. 
제 온보딩 뷰에도 비슷하게 텍스트 안에서 로직이 직접 들어간 부분이 있는데,
텍스트안에 삼항 연산자를 바로 들어가게 되면 가독성이 떨어지고 재사용도 어렵다 보니

포맷팅을 기준으로 예시를 들면,
아래와 같은 코드처럼 함수로 분리해서 처리하는 방식이 더 깔끔하다고 생각하는데 어떻게 생각하세요??
```suggestion
            Text(formatPrice())
                .applyNapzakFont(.body1Bold16)
                .foregroundStyle(Color.napzakGrayScale(.gray500))
                .frame(height: 20)
                .lineLimit(1)
                .padding(.top, 4)
                
                
extension ProductItemView {                
            private func formatPrice() -> String {
        return product.tradeType == .sell 
            ? "\(String(product.price).convertPrice(maxPrice: 1_000_000))원" 
            : "\(String(product.price).convertPrice(maxPrice: 1_000_000))원대"
    }
```

Reply:
좋습니당 반영할게요!